devtools-panels/background_scripts/background.js

```diff
@@ -0,0 +1,19 @@
+
+/**
+When we receive the message, execute the given script in the given
+tab.
+*/
+function handleMessage(request, sender, sendResponse) {
+ 
```

Comment:
I think it would be good for this to check where the message came from, especially as it's blindly executing some code in the tab. I think you should be able to do this using some code like:

```
if (sender.url != browser.runtime.getURL("/devtools/panel/panel.html")) {
  return;
}
```